Library/src/Multiplayer/SpaceTransform.cpp

```diff
@@ -25,3 +25,13 @@ csp::multiplayer::SpaceTransform::SpaceTransform(const csp::common::Vector3& Pos
 	: Position(Position), Rotation(Rotation), Scale(Scale)
 {
 }
+
+csp::multiplayer::SpaceTransform csp::multiplayer::SpaceTransform::operator+(const SpaceTransform& Transform)
+{
+	return SpaceTransform(Position + Transform.Position, Rotation + Transform.Rotation, Scale + Transform.Scale);
```

Comment:
This isn't a mathematically correct transformation from one space to another. 

Position/rotation/scale cannot be simply be added together, as each element is not just dependent upon itself, but also the other two elements.

We need to do the equivalent of a matrix multiplication here to derive a transform relative to the entity's space.

Reply:
A good reference for what converting from a (vector,quat,vector) to a matrix representation of a transform may look like is Epic's implementation [here].(https://github.com/EpicGames/UnrealEngine/blob/release/Engine/Source/Runtime/Core/Public/Math/TransformNonVectorized.h#L241). They follow a similar pattern to CSP where transform data is stored as (vector,quat,vector), and converted to a matrix when there is a need to combine two transforms together.

Note you will need to register with Epic if you haven't already in order to view engine source.

Reply:
The orientation and positions are fine to just add, but I agree the scale needs looking at.

Reply:
Sounds good